common/lib/xmodule/xmodule/tests/test_capa_module.py

```diff
@@ -361,6 +361,91 @@ def test_showanswer_past_due(self):
 
+    def test_showanswer_after_attempts_with_max(self):
+        '''
+        Cannot see answer because user has not attempted
+        answering question for the required number of
+        times.
+        '''
+        before_attempts_with_max = CapaFactory.create(
+            showanswer="after_attempts",
+            attempts="2",
+            attempts_before_showanswer_button="3",
+            max_attempts="5"
+        )
+        self.assertFalse(before_attempts_with_max.answer_available())
+
+    def test_showanswer_after_attempts_no_max(self):
+        before_attempts_no_max = CapaFactory.create(
+            showanswer="after_attempts",
+            attempts="2",
+            attempts_before_showanswer_button="3"
+        )
+        self.assertFalse(before_attempts_no_max.answer_available())
+
+    def test_showanswer_after_attempts_used_all_attempts(self):
+        '''
+        can see answer even after all attempts are used up,
+        even with the due date in the future
+        '''
+        used_all_attempts = CapaFactory.create(
+            showanswer='after_attempts',
+            attempts_before_showanswer_button="2",
+            max_attempts="3",
+            attempts="3",
+            due=self.tomorrow_str
+        )
+        self.assertTrue(used_all_attempts.answer_available())
+
+    def test_showanswer_after_attempts_past_due_date(self):
+        past_due_date = CapaFactory.create(
+            showanswer='after_attempts',
+            attempts_before_showanswer_button="2",
+            attempts="2",
+            due=self.yesterday_str
+        )
+        self.assertTrue(past_due_date.answer_available())
+
+    def test_showanswer_after_attempts_still_in_grace(self):
+        '''
+        Can see even though grace period hasn't expired as long as
+        User has already attempted for the required number of times
+        '''
+        still_in_grace = CapaFactory.create(
+            showanswer='after_attempts',
+            after_attempts="3",
+            attempts="4",
+            due=self.yesterday_str,
+            graceperiod=self.two_day_delta_str
+        )
+        self.assertTrue(still_in_grace.answer_available())
+
+    def test_showanswer_after_attempts_large(self):
+        '''
+        Ensure that if attempts_before_showanswer_button > max_attempts, the button shows up
+        after all attempts are used up, i.e after_attempts becomes max_attempts
+        '''
+        large_after_attempts = CapaFactory.create(
+            showanswer='after_attempts',
+            attempts_before_showanswer_button="5",
+            max_attempts="3",
+            attempts="3"
+        )
+        self.assertTrue(large_after_attempts.answer_available())
+
+    def test_showanswer_after_attempts_zero(self):
+        '''
+        If attempts_before_showanswer_button is 0,
+        then the show_answer button should always
+        be visible.
+        '''
+        zero_after_attempts = CapaFactory.create(
```

Comment:
Agree; I'll usually use the general term here, like `problem` or some such, that way similar lines look the same:

```python
problem = CapaFactor.create(
    # ...
)
```

This makes subsequent lines even easier to comprehend:

```python
assert problem.is_valid()
# vs
assert zero_after_attempts.is_valid()
```